agent/api/task/task.go

```diff
@@ -834,9 +841,16 @@ func (task *Task) initializeFirelensResource(config *config.Config, resourceFiel
 			if firelensConfig.Options != nil && firelensConfig.Options["enable-ecs-log-metadata"] == "false" {
 				enableECSLogMetadata = false
 			}
-
+			var networkMode string
+			if task.IsNetworkModeAWSVPC() {
+				networkMode = AWSVPCNetworkMode
+			} else if container.GetNetworkMode() == "" {
+				networkMode = BridgeNetworkMode
+			} else {
+				networkMode = container.GetNetworkMode()
```

Comment:
why won't this line work for all three cases?

Reply:
both awsvpc mode and default network mode has network mode as empty string. so we need to specially treat those two (fyi this change belongs to https://github.com/aws/amazon-ecs-agent/pull/2181)